### DISCUSSION TITLE
Creating a plugin

### DISCUSSION THREAD

**Discussion by rrd108** [vuejs/devtools#600] [Q&A] [answered]
I am trying to create a new devtools plugin and completely stucked.

And I am confused also :smile: 

So I have `vue-mess-detector` and I want to add it as a vue devtools plugin. The idea is to add a new custom tab, and when the user opens it vue-mess-detector will run display the results after the cli tool generated it.

If my understanding is correct I should add `vue-mess-detector-devtools.ts` to my `src` directory of vue-mess-detector, and use the `setupDevtoolsPlugin` function.

And then in the project where I want to use it - I should import and call it in `main.ts`

As vue-mess-detector is a cli tool, it uses node API-s, for example `process`.
So when I try to run my project it fails with `Uncaught ReferenceError: process is not defined` and it is referencing a line of the build file what is related to the cli arguments.

I would appreciate any help how to move forward with this.

**Comment by alexzhang1030** [comment, marked as answer]
Sorry for the delay, the new devtools is fully compatible with the old devtools plugin API.

https://devtools-v6.vuejs.org/plugin/plugins-guide.html
